images/autobump/autobump.sh

```diff
@@ -0,0 +1,195 @@
+set -o pipefail
+
+PROW_CONTROLLER_MANAGER_FILE="${PROW_CONTROLLER_MANAGER_FILE:-}"
+# Args for use with GH
+GH_ORG="${GH_ORG:-}"
+GH_REPO="${GH_REPO:-}"
+FORK_GH_REPO="${FORK_GH_REPO:-${GH_REPO}}"
+# Args for use with Gerrit
+GERRIT_HOST_REPO="${GERRIT_HOST_REPO:-}"
+
+# Set this to something more specific if the repo hosts multiple Prow instances.
+# Must be a valid to use as part of a git branch name. (e.g. no spaces)
+PROW_INSTANCE_NAME="${PROW_INSTANCE_NAME:-prow}"
+
+
+
+# TODO(fejta): rewrite this in a better language REAL SOON  <-lol
+main() {
+	if [[ $# -lt 1 ]]; then
+			echo "Usage: $(basename "$0") <path to github token or http cookiefile> [git-name] [git-email]" >&2
+			return 1
+	fi
+	creds=$1
+	shift
+	check-args
+	ensure-git-config "$@"
+
+	echo "Bumping ${PROW_INSTANCE_NAME} to upstream (prow.k8s.io) version..." >&2
+	/bump.sh --upstream
+
+	cd "$(git rev-parse --show-toplevel)"
+	old_version=$(git show "HEAD:${PROW_CONTROLLER_MANAGER_FILE}" | extract-version)
+	version=$(cat "${PROW_CONTROLLER_MANAGER_FILE}" | extract-version)
+
+	if [[ -z "${version}" ]]; then
+		echo "Failed to fetch version from ${PROW_CONTROLLER_MANAGER_FILE}"
+		exit 1
+	fi
+	if [[ "${old_version}" == "${version}" ]]; then
+		echo "Bump did not change the Prow version: it's still ${version}. Aborting no-op bump." >&2
+		return 0
+	fi
+	git add -u
+	title="Bump ${PROW_INSTANCE_NAME} from ${old_version} to ${version}"
+	comparison=$(extract-commit "${old_version}")...$(extract-commit "${version}")
+	body="Included changes: https://github.com/kubernetes/test-infra/compare/${comparison}"
+
+	if [[ -n "${GH_ORG}" ]]; then
+		create-gh-pr
+	else
+		create-gerrit-pr
+	fi
+
+	echo "autobump.sh completed successfully!" >&2
+}
+
+user-from-token() {
+	user=$(curl -H "Authorization: token $(cat "${token}")" "https://api.github.com/user" 2>/dev/null | sed -n "s/\s\+\"login\": \"\(.*\)\",/\1/p")
+}
+
+ensure-git-config() {
+	if [[ $# -eq 2 ]]; then
+		echo "git config user.name=$1 user.email=$2..." >&2
+		git config user.name "$1"
+		git config user.email "$2"
+	fi
+	git config user.name &>/dev/null && git config user.email &>/dev/null && return 0
+	echo "ERROR: git config user.name, user.email unset. No defaults provided" >&2
+	return 1
+}
+
+check-args() {
+	if [[ -z "${PROW_CONTROLLER_MANAGER_FILE}" ]]; then
+		echo "ERROR: $PROW_CONTROLLER_MANAGER_FILE must be specified." >&2
+		return 1
+	fi
+	if [[ -z "${GERRIT_HOST_REPO}" ]]; then
+		if [[ -z "${GH_ORG}" || -z "${GH_REPO}" ]]; then
+			echo "ERROR: GH_ORG and GH_REPO must be specified to create a GitHub PR." >&2
+			return 1
+		fi
+	else
+		if [[ -n "${GH_ORG}" || -n "${GH_REPO}" ]]; then
+			echo "ERROR: GH_ORG and GH_REPO cannot be used with GERRIT_HOST_REPO." >&2
+			return 1
+		fi
+		if [[ -z "${GERRIT_HOST_REPO}" ]]; then
+			echo "ERROR: GERRIT_HOST_REPO must be specified to create a Gerrit PR." >&2
+			return 1
+		fi
+	fi
+}
+
+create-gh-pr() {
+    gpg --import /root/gpg-signing-key/poiana.asc
```

Comment:
Can we provide these as parameters to the script/program?

Reply:
I can look into this, it should be easy for the script... for the go program might take a bit more time to figure out/test.